phase2/docker-images/toolbox/ansible/aws.yml

```diff
@@ -0,0 +1,179 @@
+    delegate_to: localhost
+    ec2_remote_facts:
+      region: "{{ ec2_instance_identity.region }}"
+      filters:
+        "instance-id": "{{ ec2_instance_identity.instanceId }}"
+    register: ec2_describe_instance
+
+  - name: Store cluster membership facts for this instance
+    set_fact:
+      kubernetes_cluster: "{{ ec2_describe_instance.instances[0].tags.KubernetesCluster }}"
+      kubernetes_role: "{{ ec2_describe_instance.instances[0].tags.Name }}"
+      pki_registry_domain: "{{ ec2_instance_identity.accountId }}.dkr.ecr.{{ ec2_instance_identity.region }}.amazonaws.com"
+      aws_local_hostname: "{{ ec2_describe_instance.instances[0].private_dns_name }}"
+      toolbox_image: "{{ lookup('env','KUBERNETES_ANYWHERE_TOOLBOX_IMAGE') }}"
+
+  - name: Fetch metadata about all other cluster members
+    delegate_to: localhost
+    ec2_remote_facts:
+      region: "{{ ec2_instance_identity.region }}"
+      filters:
+        "tag:KubernetesCluster": "{{ kubernetes_cluster }}"
+    register: ec2_describe_cluster
+
+  - name: Set cluster member list
+    set_fact:
+      kubernetes_cluster_memebers: "{{ ec2_describe_cluster.instances | map(attribute='private_ip_address') | ipv4 | list }}"
+
+  - name: Install all systemd units
+    copy: src="{{ item }}" dest="/etc/systemd/system/" owner="root" group="root" mode="0644"
+    with_fileglob:
+      - "/etc/toolbox/systemd-units/common/*"
+      - "/etc/toolbox/systemd-units/ec2/*"
+      - "/etc/toolbox/systemd-units/with-pki/*"
+
+  - file: path=/etc/kubernetes-anywhere state=directory owner="root" group="root" mode="0755"
+
+  - name: Set hostname for kubelet to use
+    copy: dest="/etc/kubernetes-anywhere/local.env"
+                mode="0644" owner="root" group="root"
+                content="AWS_LOCAL_HOSTNAME=\"{{ aws_local_hostname }}\"\nWEAVE_KNOWN_PEERS=\"{{ kubernetes_cluster_memebers | join(' ') }}\"\n"
+
+  - name: Load new systemd units
+    shell: systemctl daemon-reload
+
+  - name: Enable base systemd units
+    shell: systemctl enable \
+             weave.service \
+             weaveproxy.service \
+             weaveplugin.service \
+             weave-expose.service \
+             weave-network.target
+
+  - name: Enable base systemd units
+    shell: systemctl start weave-network.target
+
+  - name: Start ECR login service
+    when: kubernetes_role == "kubernetes-master"
+    ## as master builds PKI images, it needs to login to ECR first
+    shell: systemctl start ecr-login.service
+
+  - name: Create PKI and push to ECR
+    ## TODO: this currently runs
+    delegate_to: localhost
+    when: kubernetes_role == "kubernetes-master"
+    ## this is to be done only once, so there is no unit for it
+    docker:
+      image: "{{ toolbox_image }}"
+      detach: false
+      volumes:
+        - "/root/.docker/:/root/.docker/"
+        - "/var/run/docker.sock:/docker.sock"
+      command: "create-pki-containers" ## TODO: pass public IP
```

Comment:
I'm glad you've asked! Not exactly, it will produce new certs on re-run. Although, this code is only mean to run once, there should be a guard. I guess the solution would involve creating a file in root directory... or is there anything more specific that Ansible has? I can also do something under the hood of that command, i.e. check if a certain docker image is already present...